src/Symfony/Component/Process/Pipes/AbstractPipes.php

```diff
@@ -97,10 +101,18 @@ protected function write()
                 $input = null;
             } elseif (is_resource($input = $input->current())) {
                 stream_set_blocking($input, 0);
-            } else {
-                $this->inputBuffer .= $input;
+            } elseif (!isset($this->inputBuffer[0])) {
+                if (!is_string($input)) {
+                    if (!is_scalar($input)) {
+                        throw new InvalidArgumentException(sprintf('%s yielded a value of type "%s", but only scalars and stream resources are supported', get_class($this->input), gettype($input)));
+                    }
+                    $input = (string) $input;
```

Comment:
you're missing https://github.com/symfony/symfony/pull/18386#discussion-diff-58244419R102